src/Controller/SubscriptionController.php

```diff
@@ -136,7 +136,8 @@ public function subscribe($entity_type_id, EntityInterface $group, OgMembershipT
       return new RedirectResponse($group->toUrl()->setAbsolute(TRUE)->toString());
     }
 
-    if (!$this->ogAccess->userAccess($group, 'subscribe', $user) && !$this->ogAccess->userAccess($group, 'subscribe without approval', $user)) {
+    if ((($access = $this->ogAccess->userAccess($group, 'subscribe', $user)) && $access->isForbidden()) &&
+      (($access = $this->ogAccess->userAccess($group, 'subscribe without approval', $user)) && $access->isForbidden())) {
```

Comment:
Making this a bit more readable:
```suggestion
    $subscribe = $this->ogAccess->userAccess($group, 'subscribe', $user);
    $subscribeWithoutApproval = $this->ogAccess->userAccess($group, 'subscribe without approval', $user);
    if ($subscribe->isForbidden() && $subscribeWithoutApproval->isForbidden()) {
```